addon/components/frost-autocomplete.js

```diff
@@ -0,0 +1,325 @@
+    }
+    return value.indexOf(filterValue) !== -1
+  },
+
+  _handleArrowKey (upArrow) {
+    if (this.get('_opened') === true) {
+      this._setFocusedIndex(upArrow)
+    }
+  },
+
+  _setFocusedIndex (upArrow) {
+    const items = this.get('items')
+    if (isEmpty(items)) {
+      return
+    }
+
+    const selectedItemIndex = this.get('focusedIndex') || 0
+
+    const newSelectedIndex = (
+      upArrow ? Math.max(0, selectedItemIndex - 1) : Math.min(items.length - 1, selectedItemIndex + 1)
+    )
+
+    if (newSelectedIndex !== selectedItemIndex) {
+      this.set('focusedIndex', newSelectedIndex)
+    }
+  },
+
+  @readOnly
+  @computed('width')
+
+  /**
+   * Get a style string based on the presence of some properties
+   * @param {String} width the width property specified
+   * @returns {String} the completed style string
+   */
+  style (width) {
+    let styles = ''
+
+    // if a property is not falsy, append it to the style string
+    // note that in the width case, we want the component interface to have absolute power over the width
+    // so it will override any max or win widths to ensure ultimate control
+    if (width) styles += `width: ${width}px; max-width: initial; min-width:initial; `
+    return Ember.String.htmlSafe(styles)
+  },
+
+  @readOnly
+  @computed('disabled', 'tabIndex')
+  /**
+   * Get appropriate tab index
+   * disabled state changes.
+   * @param {Boolean} disabled - whether or not input is disabled
+   * @param {Number} tabIndex - tab index
+   * @returns {Number} tab index
+   */
+  computedTabIndex (disabled, tabIndex) {
+    return disabled ? -1 : tabIndex
+  },
+
+  @readOnly
+  @computed('opened', '_userInput', 'filter', 'focused')
+  _opened (opened, userInput, filter, focused) {
+    return (opened || focused) && userInput && !isEmpty(filter)
+  },
+
+  @readOnly
+  @computed('error')
+  /**
+   * Get appropriate class for input text
+   * @param {Boolean} error - whether or not input is disabled
+   * @returns {String} class for input text
+   */
+  inputClass (error) {
+    if (error) {
+      return 'error'
+    }
+  },
+
+  @readOnly
+  @computed('data', 'filter', 'onInput')
+  items (data, filter, onInput) {
+    if (isEmpty(data)) {
+      return []
+    }
+
+    filter = filter ? filter.toLowerCase() : null
+
+    return data.filter((item) => {
+      if (isEmpty(filter)) {
+        return true
+      }
+
+      const label = item.label || ''
+
+      if (this.isFilteredItem(label.toLowerCase(), filter)) {
+        return true
+      }
+    })
+  },
+
+  init () {
+    this._super(...arguments)
+
+    if (!isEmpty(this.selectedValue)) {
+      this.set('internalSelectedItem', {value: this.selectedValue})
+    }
+  },
+
+  didInsertElement () {
+    this._super(...arguments)
+
+    // We need jQuery instance of components root DOM node to hand off to
+    // dropdown so it can position itself properly relative to the select
+    this.set('_element', this.$())
```

Comment:
So this can be confusing. Every component has `element`. It's not obvious that `_element` is the jQuery instance of it. I would like this to be `$element`.

Reply:
@notmessenger asked me to not use $element

Reply:
Passing `$element` into a component (as a property) is not a pattern we want to encourage, though I understand what @sophypal is referring to.  It seems a more precise name is in order.